src/Paprika.Tests/Merkle/RootHashTests.cs

```diff
@@ -0,0 +1,52 @@
+using NUnit.Framework;
+using Paprika.Chain;
+using Paprika.Crypto;
+using Paprika.Merkle;
+using Paprika.Store;
+
+namespace Paprika.Tests.Merkle;
+
+public class RootHashTests
+{
+    private const int SmallDb = 256 * Page.PageSize;
+    private readonly Keccak _blockKeccak = Keccak.Compute("block"u8);
+
+    [Test]
+    public async Task Empty_database()
+    {
+        using var db = PagedDb.NativeMemoryDb(SmallDb);
+        await using var blockchain = new Blockchain(db, preCommit: new ComputeMerkleBehavior());
+
+        using var block = blockchain.StartNew(Keccak.Zero, _blockKeccak, 1);
```

Comment:
It would be nice if we could extract this into a `Setup` method

Reply:
Is it needed though? My thinking was that `IPreCommitBehavior` is sufficient to work with, at least with the calculation? It can be called with an arbitrary `ICommit` mocked implementation.

Reply:
The tests insert multiple accounts, then commit the changes in order to trigger the Merkle `IPreCommitBehavior` to recompute the root hash. For that, we actually need to store the account (and all intermediate Merkle nodes) somewhere, and for that we need this setup code.

I don't quite follow how we can get away with just a mocked `ICommit`. Could you elaborate a little?

Reply:
For now I've decided to keep it as is: the cost of avoiding the repetition is higher than repeating the same 3 lines of code. If this becomes an issue, we can revisit it.